### PR TITLE
NMS-12547: Use ProtoBuf to transport BMP messages

### PR DESCRIPTION
**Pull request by fooker**
Using a BMP-specific ProtoBuf transport format instead of BSON based
format.

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-12547

### CODE REVIEW

File: features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpPeerStatusAdapter.java [hunk 1]
```diff
@@ -77,13 +74,18 @@ public BmpPeerStatusAdapter(final AdapterDefinition adapterConfig,
     public void handleMessage(final TelemetryMessageLogEntry message,
                               final TelemetryMessageLog messageLog) {
         LOG.trace("Parsing packet: {}", message);
-        final BsonDocument document = new RawBsonDocument(message.getByteArray());
+        final Transport.Message document;
```
Comment (j-white) [inline features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpPeerStatusAdapter.java]: Should call this something other than `document` now?

File: features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpTelemetryAdapter.java [hunk 2]
```diff
@@ -99,40 +100,46 @@ public BmpTelemetryAdapter(final AdapterDefinition adapterConfig,
...
+        Optional.ofNullable(stats.getDuplicatePrefix()).ifPresent(addCounter.apply("duplicate_prefix"));
+        Optional.ofNullable(stats.getDuplicateWithdraw()).ifPresent(addCounter.apply("duplicate_withdraw"));
+        Optional.ofNullable(stats.getInvalidUpdateDueToAsConfedLoop()).ifPresent(addCounter.apply("inv_as_confed_loop"));
+        Optional.ofNullable(stats.getInvalidUpdateDueToAsPathLoop()).ifPresent(addCounter.apply("inv_as_path_loop"));
+        Optional.ofNullable(stats.getInvalidUpdateDueToClusterListLoop()).ifPresent(addCounter.apply("inv_cl_loop"));
+        Optional.ofNullable(stats.getInvalidUpdateDueToOriginatorId()).ifPresent(addCounter.apply("inv_originator_id"));
+        Optional.ofNullable(stats.getAdjRibIn()).ifPresent(addGauge.apply("adj_rib_in"));
+        Optional.ofNullable(stats.getAdjRibOut()).ifPresent(addGauge.apply("adj_rib_out"));
+
+        // TODO fooker: Add per AFI counters (perAfiAdjRibIn and perAfiLocRib)
```
Comment (j-white) [inline features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpTelemetryAdapter.java]: Worth creating a JIRA for this?

File: features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpTelemetryAdapter.java [hunk 3]
```diff
@@ -77,15 +75,18 @@ public BmpTelemetryAdapter(final AdapterDefinition adapterConfig,
     @Override
     public Stream<CollectionSetWithAgent> handleCollectionMessage(final TelemetryMessageLogEntry message,
                                                                   final TelemetryMessageLog messageLog) {
-        LOG.debug("Received {} telemetry messages", messageLog.getMessageList().size());
-
         LOG.trace("Parsing packet: {}", message);
-        final BsonDocument document = new RawBsonDocument(message.getByteArray());
+        final Transport.Message document;
```
Comment (j-white) [inline features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpTelemetryAdapter.java]: Similarly here.

File: features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpTelemetryAdapter.java (hunk 2 above)
Comment (fooker) [reply features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpTelemetryAdapter.java]: Tracking issue is https://issues.opennms.org/browse/NMS-12553

File: features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpPeerStatusAdapter.java (hunk 1 above)
Comment (fooker) [reply features/telemetry/protocols/bmp/adapter/src/main/java/org/opennms/netmgt/telemetry/protocols/bmp/adapter/BmpPeerStatusAdapter.java]: There is no good name I can come up with - `message` is already in use.

**Pull request merged**